experiments/kompanion/cmd/migrator/migrator.go

```diff
@@ -0,0 +1,369 @@
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	examples = `
+	# export Config Connector resources across all namespaces, excludes \"kube\" namespaces by default
+	kompanion migrator
+
+	# exclude certain namespace prefixes
+	kompanion migrator --exclude-namespaces=kube --exclude-namespaces=my-team
+
+	# target only specific namespace prefixes
+	kompanion migrator --target-namespaces=my-team
+
+	# target only specific namespace prefixes AND specific object prefixes
+	kompanion migrator --target-namespaces=my-team --target-objects=logging
+	`
+)
+
+func BuildMigratorCmd() *cobra.Command {
+	opts := NewMigratorOptions()
+	cmd := &cobra.Command{
+		Use:     "migrator",
+		Short:   "migrates Config Connector resources",
+		Example: examples,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunMigrator(cmd.Context(), opts)
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
+
+	return cmd
+}
+
+// Task is implemented by our namespace-collection routine, or anything else we want to run in parallel.
+type Task interface {
+	Run(ctx context.Context) error
+}
+
+// tracks the namespaces to be exported.
+// thread safe.
+type taskQueue struct {
+	mu    sync.Mutex
+	tasks []Task // will be treated as a FIFO queue
+}
+
+func (n *taskQueue) GetWork() Task {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	if len(n.tasks) == 0 {
+		return nil
+	}
+
+	workItem := n.tasks[0]
+	n.tasks = n.tasks[1:]
+
+	return workItem
+}
+
+func (n *taskQueue) AddTask(t Task) {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	n.tasks = append(n.tasks, t)
+}
+
+type dumpResourcesTask struct {
+	// Namespace is the namespace to filter down
+	Namespace string
+
+	DynamicClient *dynamic.DynamicClient
+
+	OutputChannel chan string
+
+	// Resources is the list of resources to query
+	Resources []schema.GroupVersionResource
+
+	ShouldExcludeObject func(id types.NamespacedName) bool
+}
+
+func (t *dumpResourcesTask) Run(ctx context.Context) error {
+	for _, gvr := range t.Resources {
+		var resources *unstructured.UnstructuredList
+		if t.Namespace != "" {
+			r, err := t.DynamicClient.Resource(gvr).Namespace(t.Namespace).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("fetching gvr %s resources in namespace %s: %w", gvr, t.Namespace, err)
+			}
+			resources = r
+
+		} else {
+			r, err := t.DynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("fetching gvr %s resources: %w", gvr, err)
+			}
+			resources = r
+		}
+
+		for _, r := range resources.Items {
+			id := types.NamespacedName{
+				Namespace: r.GetNamespace(),
+				Name:      r.GetName(),
+			}
```

Comment:
I think in the future we would want to support a wf where we migrate all the kinds of a resource. So down the road it may be worth building out a kind based filter too (like we have around the namespace and the name of the obj today).